### PR TITLE
Update registration process to reflect 3 tables

### PR DESCRIPTION
**Pull request by sidpremkumar**
## Description of the change

> Description here (why is it needed, what does it do)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [ ] Enhancement (minor change below the level of a feature)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-1014

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

### CODE REVIEW

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go [hunk 1]
```diff
@@ -1,27 +1,32 @@
...
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
 const (
-	hubEndpoint   = "/api/v1/hub/kube-control"
-	autoReconnect = true
+	hubEndpoint       = "/api/v1/hub/kube-control"
+	registerEndpoint  = "/api/v1/kube/register-agent"
+	challangeEndpoint = "/api/v1/kube/get-challange"
```
Comment (tflpd) [inline bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go:28]: pls dont hate me, typo in challenge

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go [hunk 2]
```diff
@@ -41,17 +46,29 @@ func NewControlChannel(serviceUrl string,
 	environmentId string,
 	agentVersion string,
 	targetSelectHandler func(msg wsmsg.AgentMessage) (string, error)) (*ControlChannel, error) {
+	// populate keys if they haven't been generated already
+	// TODO: revisit this
```
Comment (tflpd) [inline bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go:50]: Why revisit this?

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go [hunk 3]
```diff
@@ -173,13 +231,15 @@ func aliveCheck() ([]byte, error) {
 	return aliveBytes, nil
 }
 
-func newAgent() error {
+func newAgent(serviceUrl string, activationToken string, agentVersion string, orgId string, environmentId string, clusterName string) (*vault.Vault, error) {
 	config, _ := vault.LoadVault()
 
 	// Check if vault is empty, if not generate a private, public key pair
```
Comment (tflpd) [inline bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go:237]: I think comment this is wrong? We gen keypair if it is empty

File: bctl-go-daemon/bctl/daemon/plugin/kube/kube.go [hunk 4]
```diff
@@ -77,6 +83,15 @@ func NewKubeDaemonPlugin(localhostToken string, daemonPort string, certPath stri
...
 	}()
 
+	go func() {
+		for {
+			select {
+			case doneMessage := <-plugin.DoneChannel:
+				plugin.ExitMessage = doneMessage
+			}
+		}
+	}()
```
Comment (tflpd) [inline bctl-go-daemon/bctl/daemon/plugin/kube/kube.go:93]: Should we add a context cancel for this go routine? Otherwise it will never exit (?)

**Review by tflpd — APPROVED** [review]

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go (hunk 2 above)
Comment (sidpremkumar) [reply bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go:50]: I dont think its needed, removed it

File: bctl-go-daemon/bctl/daemon/plugin/kube/kube.go (hunk 4 above)
Comment (sidpremkumar) [reply bctl-go-daemon/bctl/daemon/plugin/kube/kube.go:92]: Added cancel context to both go routines

Comment (sidpremkumar) [reply bctl-go-daemon/bctl/daemon/plugin/kube/kube.go]: https://github.com/bastionzero/zli/pull/152/commits/ceda376e64093c1cd1f8138f3aee69bc3a74cb2b

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go (hunk 3 above)
Comment (sidpremkumar) [reply bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go]: https://github.com/bastionzero/zli/pull/152/commits/ceda376e64093c1cd1f8138f3aee69bc3a74cb2b

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go (hunk 2 above)
Comment (sidpremkumar) [reply bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go]: https://github.com/bastionzero/zli/pull/152/commits/ceda376e64093c1cd1f8138f3aee69bc3a74cb2b

File: bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go (hunk 1 above)
Comment (sidpremkumar) [reply bctl-go-daemon/bctl/agent/controlchannel/controlchannel.go]: https://github.com/bastionzero/zli/pull/152/commits/ceda376e64093c1cd1f8138f3aee69bc3a74cb2b

**Pull request merged**